File: zadacha7.c
#include <stdio.h>

int main()
{
 const int m=5, n=5;
 int a[m][n];
 int i, j, jmin, imax, jmax;
 
 srand(time(NULL));
 
 for (i=0;i<n; i++)
 {
  for (j=0; j<m; j++)
  {
    a[i][j]=rand() % 30;
    printf("%4d ", a[i][j]);
  }
 printf("\n");
 }
 
 imax = 1;
 jmax = 1;
 
 for (j=0;j<n; j++)
   if (a[1][jmax] > a[1][j])
   {
     jmax = j;
   }
   for (i=0;i<n; i++)
   {
     jmin = 1;
     for (j=0;j<n; j++)
       if (a[i][jmin] > a[i][jmin])
       {
	 imax = i;
	 jmax = jmin;
       }
   }
     
 /*for (i=0;i<m; i++)
 {
  for (j=0; j<n; j++)
  {
   a[i][j]=rand() % 30;
   printf("%4d ", a[i][j]);
  }
  printf("\n");
 }
 printf("\n");
 
   imax = 1;
   jmax = 1;
   
   for (j=0;j<n;j++)
    if (a[imax][jmax]>a[i][j])
     jmax=j;
     for (i=0;j<m;i++)
     {
      jmin=1;
      for (j=0;j<n;j++)
      if (a[i][jmin]>a[i][j])
      jmin=j;
      if (a[imax][jmax]<a[i][jmin])
      {
       imax=i;
       jmax=jmin;
      }
     }*/
     printf("Максимальный среди минимальных в позиции: %d,%d\n", imax, jmax);
     return 0;
}

File: zadacha7.pas
const
  m=5; n=5; LIM=30; W=4;
var
  a: array [1..m,1..n] of Real;
  i, j, jmin, imax, jmax: Integer;
begin
  Randomize;
  for i:=1 to m do 
   for j:=1 to n do 
   a[i,j]:=Random(LIM);
    for i:=1 to m do 
    begin 
     for j:=1 to n do 
      Write(a[i,j]:W:0); 
      Writeln; 
    end;
  imax:=1; 
  jmax:=1; 
  for j:=2 to n do 
   if a[1,jmax]>a[1,j] then 
   jmax:=j;
   for i:=2 to m do 
   begin
    jmin:=1; 
    for j:=2 to n do 
     if a[i,jmin]>a[i,j] then 
     jmin:=j;
      if a[imax,jmax]<a[i,jmin] then 
      begin
       imax:=i; 
       jmax:=jmin;
      end;
   end;
  WriteLn('Максимальный среди минимальных в позиции: ',imax,',',jmax);
end.